barretenberg/cpp/src/barretenberg/vm2/tracegen/lib/phase_spec.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include <cstdint>
+
+namespace bb::avm2::tracegen {
+
+class TxPhaseOffsetsTable {
+
+  public:
+    struct Offsets {
+        uint32_t read_pi_offset;
+        uint32_t write_pi_offset;
+        uint32_t read_pi_length_offset;
+    };
+
+    static const TxPhaseOffsetsTable& get();
+
+    const Offsets& get_offsets(TransactionPhase phase) const;
```

Comment:
You could make this static. Non-static is also ok but I've done it mostly with things in simulation so that they can be mocked.